File: php/secret-handshake/SecretHandshake.php
<?php declare(strict_types=1);

class SecretHandshake
{
    public function commands(int $handshake): array
    {
        $instructions = [];

        if (0b00001 & $handshake) {
            $instructions[] = "wink";
        }

        if (0b00010 & $handshake) {
            $instructions[] = "double blink";
        }

        if (0b00100 & $handshake) {
            $instructions[] = "close your eyes";
        }

        if (0b01000 & $handshake) {
            $instructions[] = "jump";
        }

        if (0b10000 & $handshake) {
            $instructions = array_reverse($instructions);
        }

        return $instructions;
    }
}

File: python/secret-handshake/secret_handshake.py
def commands(binary_str):
    input = int(binary_str, 2)
    output = []

    if input & 0b1:
        output.append("wink")

    if input & 0b10:
        output.append("double blink")

    if input & 0b100:
        output.append("close your eyes")

    if input & 0b1000:
        output.append("jump")

    if input & 0b10000:
        output = output[::-1]

    return output
